### PR TITLE
fix(cli): resolve verified mutation bugs in add/move/edit/import/sync/repair-upstream/export

### PR DESCRIPTION
**Pull request by mfacenet**
## Summary

Fixes 9 verified code-review findings across the CLI mutation commands (`add`, `move`, `edit`, `import`, `sync`, `repair-upstream`, `export`), plus dead-code cleanup where safe to do in-scope. One finding is partially deferred — see "Deferred" below.

## Findings fixed

1. **P1** `add.go`/`move.go` — `filepath.Abs(filepath.Join(cwd, target))` re-rooted an absolute target path under `cwd` (`filepath.Join` doesn't special-case absolute segments, so `Join("/cwd", "/abs/target")` → `/cwd/abs/target`). Added `resolveAbsoluteTargetPath`, mirroring `describe.go`'s absolute-first pattern: use an absolute target as-is (cleaned), otherwise join with cwd.
   - Tests: `TestResolveAbsoluteTargetPath`, `TestAddCommandWithAbsoluteTargetDoesNotReRootUnderCWD`, `TestMoveCommandWithAbsoluteTargetDoesNotReRootUnderCWD`.

2. **P1** `edit.go:202-208` `trackingBranchFromUpstream` split on `/` and returned the last segment, so `origin/feature/x` collapsed to `x`; `repair-upstream` then rewired tracking to `origin/x`. Fixed to strip only the first segment (the remote) via `strings.Cut`.
   - Test: `TestTrackingBranchFromUpstream` (table-driven, includes multi-segment branch case).

3. **P1** `import.go:112-134` — in merge mode, `--include-registry=false` still computed clone entries from the bundle and called `ExecuteImportClones`, which upserts every cloned entry into `cfg.Registry` regardless of `includeRegistry` — only `--file-only` previously suppressed cloning. Gated merge-mode cloning on `includeRegistry` explicitly (replace mode was already safe via `cfg.Registry == nil`).
   - Test: `TestImportCommandRunEMergeIncludeRegistryFalseDoesNotCloneOrRegisterBundleRepos` (verified it fails without the fix).

4. **P2** `sync.go:118-131` — `--dry-run` still ran the interactive confirmation before the dry-run branch, so a non-interactive dry-run (piped stdin, `-o json` in CI) hit EOF, was treated as declined, and exited without printing the plan. Confirmation is now skipped entirely when `--dry-run` is set.
   - Test: `TestSyncRunEDryRunSkipsConfirmationPrompt` (verified it fails without the fix).

5. **P2** `sync.go:316-326` `syncResultNeedsConfirmation` gated local rebase/stash/clone but not `git push`, so `--update-local --push-local` pushed to remotes with no prompt. Added `git push` to the gate.
   - Test: added cases to the existing `TestSyncResultNeedsConfirmationTable`.

6. **P2** `sync.go` — a successful `--checkout-missing` clone only updated the engine's in-memory registry; `sync` RunE never called `config.Save`, so the next run re-planned and re-attempted the same clone indefinitely. Registry is now persisted after a non-dry-run sync when any checkout-missing clone succeeded.
   - Test: `TestSyncRunEPersistsRegistryAfterCheckoutMissingClone` (verified it fails without the fix).

7. **P2** `edit.go:189-198` `validateEditedRegistryEntry` rejected an edit whenever *any* other entry shared its `repo_id`, breaking the supported multi-checkout case. Uniqueness is now keyed on `(repo_id, checkout_id)` when set, or `(repo_id, path)` otherwise.
   - Test: `TestValidateEditedRegistryEntryUniqueness` (table-driven).

8. **P2** `repair_upstream.go:271-282` `repairUpstreamMatchesFilter`'s default arm returned `true`, so an unrecognized `--only` value silently meant "all". Added `parseUpstreamRepairFilter` to validate `--only` up front (mirroring `parseImportMode`/`parseOutputMode`); `repairUpstreamMatchesFilter` itself is unchanged (its default is now unreachable from RunE).
   - Tests: `TestParseUpstreamRepairFilter`, `TestRepairUpstreamRunERejectsUnknownOnlyFilter`.

9. **P2** `export.go:91` wrote the bundle `0o644` with `entry.RemoteURL` verbatim, so credential URLs (`https://user:token@host/...`) leaked into a world-readable file. Now writes `0o600`; also warns (naming affected repo IDs) whenever an exported `remote_url` carries embedded credentials, reusing `urlutil.RedactCredentials` as the detector. `remote_url` is **not** redacted in the bundle body itself, since it must stay usable for `import`/`reconcile`.
   - Tests: `TestExportCommandRunEWritesFileOwnerOnlyPermissions`, `TestExportCommandRunEWarnsOnEmbeddedCredentials`, `TestExportEntriesWithEmbeddedCredentials`.

## Dead code (finding #10)

`sync.go:537` `writeSyncTable`'s `report *model.StatusReport` parameter and `sync.go:481-487` `writeProgressLine`'s `newline` param in the non-TTY branch:

- **`writeProgressLine`**: both non-TTY branches wrote the identical `"%s\n"` line, so `newline` had no effect there. **Removed** — collapsed to one statement (pure dedup, no behavior change; `newline` still matters in the in-place TTY branch above it).
- **`writeSyncTable`'s `report` param**: always `nil` at both call sites in `sync.go` (`:182`, `:187`), including via the `reconcile` alias (`kubectl_aliases.go` sets `reconcileCmd.RunE = syncCmd.RunE`), so the status-enrichment block and the `sync -o wide` `PRIMARY_REMOTE`/`UPSTREAM`/`AHEAD`/`BEHIND` columns never populate in production today. **Left as-is** for two reasons:
  1. The function's signature is exercised directly (with a non-nil report) by `helpers_test.go` and `terminal_width_test.go`, which are outside this PR's file scope — removing the parameter requires editing those.
  2. Wiring in a *real* report means running full per-repo `InspectRepo` on every plain `sync` call (today that only happens conditionally, e.g. under `--update-local` or certain `--only` filters), which is a behavior/perf change beyond a mutation-bug fix and needs its own decision, not a guess buried in this PR.

  A maintainer should decide: (a) wire `sync`/`reconcile` to build and pass a real `StatusReport` (accepting the added per-repo inspection cost), or (b) remove the dead parameter/columns outright (touching the two test files above). Flagging for follow-up rather than guessing.

## Tests added

Table-driven where the existing convention uses it; RunE-level regression tests reproduce each bug end-to-end and were confirmed to fail on the pre-fix code (verified for findings #1, #3, #4, #6, #9 by temporarily reverting each fix and re-running).

## Verification

```
go build ./...                                          # pass
go vet ./cmd/repokeeper/...                              # pass
go run honnef.co/go/tools/cmd/staticcheck@v0.7.0 ./cmd/repokeeper/...   # pass
go test ./cmd/repokeeper/...                              # pass
go test -race ./cmd/repokeeper/...                        # pass, except a pre-existing, unrelated
                                                           # data race in internal/registry.backfillCheckoutID
                                                           # / internal/engine.InspectRepo, reachable via
                                                           # status.go's TestStatusRunE* tests. Reproduced this
                                                           # on a clean stash of origin/main (no changes from this
                                                           # PR) to confirm it predates this work — it's outside
                                                           # this PR's file scope (status.go, internal/engine,
                                                           # internal/registry) and looks like it belongs to a
                                                           # sibling in-flight PR touching status.go/scan.go.
                                                           # `go test -race ./cmd/repokeeper/... -skip TestStatusRunE`
                                                           # passes clean.
```

## Scope note

Per the task's file scope, only `cmd/repokeeper/{add,move,edit,import,sync,repair_upstream,export}.go` were changed for production code. Adding required test coverage unavoidably touched a few shared test files outside that literal list (none of the 7 production files have their own dedicated test file for every fix, and some of the functions being fixed already had table-driven tests living in shared files):

- `cmd/repokeeper/command_parsing_test.go` — appended 2 cases to the existing `TestSyncResultNeedsConfirmationTable` (git-push).
- `cmd/repokeeper/command_run_test.go` — added 2 new RunE-level sync tests (dry-run confirmation, checkout-missing persistence).
- `cmd/repokeeper/edit_test.go`, `export_test.go`, `import_test.go`, `move_test.go`, `repair_upstream_test.go` — each is the dedicated test file for one of the 7 production files; extended in place.
- `cmd/repokeeper/add_test.go` — new file (none existed for `add.go`).

None of these touch `status.go`, `scan.go`, `kubectl_aliases.go`, or their dedicated test files (`status_characterization_test.go`, `status_helpers_test.go`, `kubectl_aliases_test.go`), which are reserved for the sibling PR.

### CODE REVIEW

File: cmd/repokeeper/export.go [hunk 1]
```diff
@@ -88,7 +96,9 @@ var exportCmd = &cobra.Command{
 		if err := os.MkdirAll(filepath.Dir(outputPath), 0o755); err != nil {
 			return err
 		}
-		if err := os.WriteFile(outputPath, data, 0o644); err != nil {
+		// Bundle files can carry remote_url credentials (see the warning above),
+		// so write owner-only rather than the previous world-readable 0o644.
+		if err := os.WriteFile(outputPath, data, 0o600); err != nil {
 			return err
 		}
```
[the remark this answers is not in the document]
Comment (mfacenet) [reply cmd/repokeeper/export.go:109]: Fixed in 757e599. You're right — `os.WriteFile`'s perm bits only apply on create, so re-exporting over an existing `0o644` bundle kept it world-readable. The export now `os.Remove`s any existing file first (ignoring `os.IsNotExist`), so the write always creates a fresh `0o600` file — no window where new credential-bearing content sits at the old mode.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: cmd/repokeeper/command_run_test.go [hunk 2]
```diff
@@ -561,6 +561,144 @@ func TestSyncRunEUnsupportedFormat(t *testing.T) {
...
+	// Deliberately empty stdin: a prompt read here hits EOF immediately.
+	syncCmd.SetIn(bytes.NewReader(nil))
+	defer syncCmd.SetOut(os.Stdout)
+	defer syncCmd.SetErr(os.Stderr)
+	defer syncCmd.SetIn(nil)
+
+	_ = syncCmd.Flags().Set("only", "all")
+	_ = syncCmd.Flags().Set("dry-run", "true")
+	_ = syncCmd.Flags().Set("yes", "false")
+	_ = syncCmd.Flags().Set("checkout-missing", "true")
```
[the remark this answers is not in the document]
Comment (mfacenet) [reply cmd/repokeeper/command_run_test.go:612]: Fixed in 757e599. `yes` is a root persistent flag (`rootCmd.PersistentFlags()`), so `syncCmd.Flags().Set("yes", ...)` silently failed and left the test dependent on a `--yes` value leaked by an earlier test. It now captures `prevYes`, sets `yes=false` on `rootCmd.PersistentFlags()`, and restores it in the defer — matching the pattern the rest of the suite already uses (e.g. `import_test.go`). Confirmed the test still passes.

---
_Generated by [Claude Code](https://claude.ai/code)_
